VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -171,6 +172,24 @@ public void onDestroy() {
         EventBus.getDefault().unregister(this);
     }
 
+    @Override
+    public void onRestart() {
+        if (postsSort == DiscussionPostsSort.LAST_ACTIVITY_AT && mSelectedItem != -1) {
+            // Move the last viewed thread to the top
+            discussionPostsAdapter.moveToTop(mSelectedItem);
+            mSelectedItem = -1;
+        }
+
+        nextPage = 1;
+        controller.refresh();
```

Comment:
The implementation of the `onRestart()` callback is not appropriate here, as we only want to reload the list on the event of the user returning from the item detail Activity, not in the case of any and all restarts. Therefore, I guess we can do this from the `onActivityResult()` callback, and start the details Activity via the `startActivityForResult()` method with a non-negative request code.